apps/system/js/permission_manager.js

```diff
@@ -338,6 +338,7 @@
     handlePermissionPrompt: function pm_handlePermissionPrompt(detail) {
       // Clean dialog if was rendered before
       this.cleanDialog();
+      this.isFullscreenRequest = false;
```

Comment:
Reason is that cleanDialog logic is different if it's fullscreen or not, and I would like to have this path simple, due to it's going to land in 2.2 as well.
Maybe it's a good idea to clear some of the methods, but let's do it in master in a separate bug :). Thanks for the review!